app/domain/filter_params.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class FilterParams
+  def initialize(params)
+    @params = params
+  end
+
+  def filters_and_results
+    filters = search_filters
+    results = search_results(filters.filter { |f| f.exclude?(nil) })
+    [filters, results]
+  end
+
+  private
+
+  def search_filters
+    unless skill_ids && levels && interests
```

Comment:
make it a one liner
`return [default_filter] unless skill_ids && levels && interests`